p2panda-net/src/network.rs

```diff
@@ -176,9 +178,15 @@ where
     /// Sets the sync protocol for this network.
     ///
     /// If a sync protocol is provided, a sync session will be completed with any known peers with
-    /// whom we share topics of interest.
-    pub fn sync(mut self, protocol: impl for<'a> SyncProtocol<'a, T> + 'static) -> Self {
+    /// whom we share topics of interest. The resync flag determines whether sync should only be
+    /// completed once or repeated at regular intervals.
+    pub fn sync(
+        mut self,
+        protocol: impl for<'a> SyncProtocol<'a, T> + 'static,
+        resync: bool,
```

Comment:
Would be cool to pass a whole `SyncConfiguration` struct here, so we can add further options in the future. There's already a few others which are currently hard-coded which could live here.

Reply:
Ya, good idea!